.github/workflows/build.yml

```diff
@@ -1,19 +1,22 @@
-# A simple workflow to run tests using pytest against supported Python versions
+# Build OpenAssetIO, publish it, and test is, all in one handy workflow.
 
-name: Tests
+name: Build
 on:
   pull_request:
     branches-ignore:
       - docs
     paths-ignore:
       - 'docs/**'
+  push:
+    branches:
+      - main
```

Comment:
Yeah it was. It's a consequence of this approach. The concern was that we're not testing the specific artifact we're uploading. We want main to have runs with the artifacts so we can pull the latest version of an artifact from downstream repos, thus we should test em.

Even without that, not running tests on main would require further rework of the CI that i'd like to avoid.

Reply:
No real harm? Its essentially redundant - but I like to think of it as "test the build we just made"